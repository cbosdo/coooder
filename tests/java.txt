/**
 * Highlighter implementation using the GeSHi algorithm adapted to LibreOffice API.
 * 
 * @author cbosdonnat
 */
public final class HighlighterImpl extends WeakBase implements XServiceInfo, XHighlighter {
    
    private static final String IMPLEMENTATION_NAME = HighlighterImpl.class.getName();
    private static final String[] SERVICE_NAMES = { "org.libreoffice.coooder.Highlighter" };
    
    private static final int OPEN = 0;
    private static final int CLOSE = 1;
    
    private static final String NUMBER_REGEX = "[-+]?\\b(?:[0-9]*\\.)?[0-9]+\\b";
    
    
    private final XComponentContext mContext;
    private XLanguage mLanguage;
    
    private CompiledPatternsCache mPatternsCache;

    private XTextRange mSelectionStart;
    private XTextDocument mTextDocument;
    private int mLength;
    

    public HighlighterImpl(XComponentContext pContext) {
        mContext = pContext;
        mPatternsCache = new CompiledPatternsCache();
    };

    public static XSingleComponentFactory __getComponentFactory(String pImplementationName) {
        XSingleComponentFactory xFactory = null;

        if (pImplementationName.equals(IMPLEMENTATION_NAME)) {
            xFactory = Factory.createComponentFactory(HighlighterImpl.class, SERVICE_NAMES);   
        }
        return xFactory;
    }
}

