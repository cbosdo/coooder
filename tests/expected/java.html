<pre class="java java" style="font-family:monospace;"><span style="color: #008000; font-style: italic; font-weight: bold;">/**
 * Highlighter implementation using the GeSHi algorithm adapted to LibreOffice API.
 * 
 * @author cbosdonnat
 */</span>
<span style="color: #000000; font-weight: bold;">public</span> <span style="color: #000000; font-weight: bold;">final</span> <span style="color: #000000; font-weight: bold;">class</span> HighlighterImpl <span style="color: #000000; font-weight: bold;">extends</span> WeakBase <span style="color: #000000; font-weight: bold;">implements</span> XServiceInfo, XHighlighter <span style="color: #009900;">&#123;</span>
&nbsp;
    <span style="color: #000000; font-weight: bold;">private</span> <span style="color: #000000; font-weight: bold;">static</span> <span style="color: #000000; font-weight: bold;">final</span> <a href="http://www.google.com/search?hl=en&amp;q=allinurl%3Astring+java.sun.com&amp;btnI=I%27m%20Feeling%20Lucky"><span style="color: #003399;">String</span></a> IMPLEMENTATION_NAME <span style="color: #339933;">=</span> HighlighterImpl.<span style="color: #000000; font-weight: bold;">class</span>.<span style="color: #006633;">getName</span><span style="color: #009900;">&#40;</span><span style="color: #009900;">&#41;</span>;
    <span style="color: #000000; font-weight: bold;">private</span> <span style="color: #000000; font-weight: bold;">static</span> <span style="color: #000000; font-weight: bold;">final</span> <a href="http://www.google.com/search?hl=en&amp;q=allinurl%3Astring+java.sun.com&amp;btnI=I%27m%20Feeling%20Lucky"><span style="color: #003399;">String</span></a><span style="color: #009900;">&#91;</span><span style="color: #009900;">&#93;</span> SERVICE_NAMES <span style="color: #339933;">=</span> <span style="color: #009900;">&#123;</span> <span style="color: #0000ff;">&quot;org.libreoffice.coooder.Highlighter&quot;</span> <span style="color: #009900;">&#125;</span>;
&nbsp;
    <span style="color: #000000; font-weight: bold;">private</span> <span style="color: #000000; font-weight: bold;">static</span> <span style="color: #000000; font-weight: bold;">final</span> <span style="color: #000066; font-weight: bold;">int</span> OPEN <span style="color: #339933;">=</span> 0;
    <span style="color: #000000; font-weight: bold;">private</span> <span style="color: #000000; font-weight: bold;">static</span> <span style="color: #000000; font-weight: bold;">final</span> <span style="color: #000066; font-weight: bold;">int</span> CLOSE <span style="color: #339933;">=</span> <span style="color: #cc66cc;">1</span>;
&nbsp;
    <span style="color: #000000; font-weight: bold;">private</span> <span style="color: #000000; font-weight: bold;">static</span> <span style="color: #000000; font-weight: bold;">final</span> <a href="http://www.google.com/search?hl=en&amp;q=allinurl%3Astring+java.sun.com&amp;btnI=I%27m%20Feeling%20Lucky"><span style="color: #003399;">String</span></a> NUMBER_REGEX <span style="color: #339933;">=</span> <span style="color: #0000ff;">&quot;[-+]?<span style="color: #000099; font-weight: bold;">\\</span>b(?:[0-9]*<span style="color: #000099; font-weight: bold;">\\</span>.)?[0-9]+<span style="color: #000099; font-weight: bold;">\\</span>b&quot;</span>;
&nbsp;
&nbsp;
    <span style="color: #000000; font-weight: bold;">private</span> <span style="color: #000000; font-weight: bold;">final</span> XComponentContext mContext;
    <span style="color: #000000; font-weight: bold;">private</span> XLanguage mLanguage;
&nbsp;
    <span style="color: #000000; font-weight: bold;">private</span> CompiledPatternsCache mPatternsCache;
&nbsp;
    <span style="color: #000000; font-weight: bold;">private</span> XTextRange mSelectionStart;
    <span style="color: #000000; font-weight: bold;">private</span> XTextDocument mTextDocument;
    <span style="color: #000000; font-weight: bold;">private</span> <span style="color: #000066; font-weight: bold;">int</span> mLength;
&nbsp;
&nbsp;
    <span style="color: #000000; font-weight: bold;">public</span> HighlighterImpl<span style="color: #009900;">&#40;</span>XComponentContext pContext<span style="color: #009900;">&#41;</span> <span style="color: #009900;">&#123;</span>
        mContext <span style="color: #339933;">=</span> pContext;
        mPatternsCache <span style="color: #339933;">=</span> <span style="color: #000000; font-weight: bold;">new</span> CompiledPatternsCache<span style="color: #009900;">&#40;</span><span style="color: #009900;">&#41;</span>;
    <span style="color: #009900;">&#125;</span>;
&nbsp;
    <span style="color: #000000; font-weight: bold;">public</span> <span style="color: #000000; font-weight: bold;">static</span> XSingleComponentFactory __getComponentFactory<span style="color: #009900;">&#40;</span><a href="http://www.google.com/search?hl=en&amp;q=allinurl%3Astring+java.sun.com&amp;btnI=I%27m%20Feeling%20Lucky"><span style="color: #003399;">String</span></a> pImplementationName<span style="color: #009900;">&#41;</span> <span style="color: #009900;">&#123;</span>
        XSingleComponentFactory xFactory <span style="color: #339933;">=</span> <span style="color: #000066; font-weight: bold;">null</span>;
&nbsp;
        <span style="color: #000000; font-weight: bold;">if</span> <span style="color: #009900;">&#40;</span>pImplementationName.<span style="color: #006633;">equals</span><span style="color: #009900;">&#40;</span>IMPLEMENTATION_NAME<span style="color: #009900;">&#41;</span><span style="color: #009900;">&#41;</span> <span style="color: #009900;">&#123;</span>
            xFactory <span style="color: #339933;">=</span> Factory.<span style="color: #006633;">createComponentFactory</span><span style="color: #009900;">&#40;</span>HighlighterImpl.<span style="color: #000000; font-weight: bold;">class</span>, SERVICE_NAMES<span style="color: #009900;">&#41;</span>;   
        <span style="color: #009900;">&#125;</span>
        <span style="color: #000000; font-weight: bold;">return</span> xFactory;
    <span style="color: #009900;">&#125;</span>
<span style="color: #009900;">&#125;</span></pre>

